dotCMS/src/main/java/com/dotcms/cms/login/LoginServiceAPIImpl.java

```diff
@@ -0,0 +1,537 @@
+        return true;
+    }
+
+
+    /**
+     * Return the current login user. A {@link UserLoggingException} is thrown if any error happened
+     * getting the user.
+     *
+     * @param req
+     * @return login user
+     */
+    public User getLoggedInUser(HttpServletRequest req) {
+        User user = null;
+
+        if (req != null) {
+            try {
+                user = userWebAPI.getLoggedInUser(req);
+            } catch (PortalException | SystemException e) {
+                throw new UserLoggingException(e);
+            }
+        }
+        return user;
+    }
+
+    public User getLoggedInUser() {
+        HttpServletRequest request = this.httpServletRequestThreadLocal.getRequest();
+        return this.getLoggedInUser(request);
+    }
+
+
+    private void syncPassword(User user) {
+        try {
+
+            final boolean SYNC_PASSWORD = BaseAuthenticator.SYNC_PASSWORD;
+            if (!SYNC_PASSWORD) {
+
+                String roleName = LDAPImpl.LDAP_USER_ROLE;
+
+                if (APILocator.getRoleAPI().doesUserHaveRole(user, roleName)) {
+
+                    user.setPassword(DotCustomLoginPostAction.FAKE_PASSWORD);
+                    APILocator.getUserAPI().save(user, systemUser, false);
+                }
+            }
+        } catch (Exception e) {
+
+            Logger.debug(this.getClass(), "syncPassword not set or unable to load user", e);
+        }
+    }
+
+    private int runPreAuth(final String userName, final String password, final Company comp)
+                    throws AuthException {
+
+        if (UtilMethods.isSet(preAuthenticator) && LicenseUtil.getLevel() > 100) {
+            Authenticator authenticator =
+                            (Authenticator) ReflectionUtils.newInstance(preAuthenticator);
+
+            if (authByEmail) {
+                return authenticator.authenticateByEmailAddress(comp.getCompanyId(), userName,
+                                password);
+            } else {
+                return authenticator.authenticateByUserId(comp.getCompanyId(), userName, password);
+            }
+
+        }
+        return Authenticator.DNE;
+
+    }
+
+    private User _loadUserByKey(String key)
+                    throws DotDataException, DotSecurityException, NoSuchUserException {
+        // Search for the system user
+
+        User user = (authByEmail) ? userAPI.loadByUserByEmail(key, systemUser, false)
+                        : userAPI.loadUserById(key, systemUser, false);
+
+        if (systemUser.equals(user)) {
+            throw new LoginSecurityException(
+                            "An invalid attempt to login as a System User has been made  - you cannot login as the System User");
+        }
+
+        if (!user.isActive()) {
+            throw new LoginSecurityException(
+                            "User not active" + user.getUserId() + " " + user.getFullName());
+        }
+        if ((user == null) || (!UtilMethods.isSet(user.getEmailAddress()))) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this expression which always evaluates to "false" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2589)